nodecg-io-serial/extension/SerialClient.ts

```diff
@@ -0,0 +1,92 @@
+import { ServiceClient } from "nodecg-io-core/extension/types";
+import { success, error, Result } from "nodecg-io-core/extension/utils/result";
+import { ReadLine } from "readline";
+import SerialPort = require("serialport"); // This is neccesary, because serialport only likes require!
+// This is neccesary because serialport does not work with import...
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const Readline = require("@serialport/parser-readline");
+
+export interface DeviceInfo {
+    port: string;
+    manucaturer: string;
+    serialNumber: string;
+    pnpId: string;
+}
+
+interface Protocoll {
+    delimiter: string;
+    encoding: string;
+}
+
+export interface SerialServiceConfig {
+    device: DeviceInfo;
+    connection: SerialPort.OpenOptions;
+    protocoll: Protocoll;
+}
+
+export class SerialServiceClient implements ServiceClient<SerialPort> {
+    private serialPort: SerialPort;
+    private parser: ReadLine;
+    constructor(settings: SerialServiceConfig) {
+        SerialServiceClient.inferPort(settings.device).then((port) => {
+            if (!port.failed) {
+                this.serialPort = new SerialPort(port.result, settings.connection);
+                this.parser = this.serialPort.pipe(new Readline(settings.protocoll));
+            }
+        });
+    }
+
+    getNativeClient(): SerialPort {
+        return this.serialPort;
+    }
+
+    static async inferPort(deviceInfo: DeviceInfo): Promise<Result<string>> {
+        const result = [];
+        const devices = await SerialPort.list();
+        if ("port" in deviceInfo) {
+            result.push(deviceInfo.port);
+        } else {
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            devices.forEach((element: any) => {
```

Comment:
```suggestion
            devices.forEach((element) => {
```
The element is of type `SerialPort.PortInfo` but TypeScript will infer that type automatically when we provide none.

Reply:
This was a leftover from a much earlier version...
The compiler was complaining that the compiler was unable to properly infer the type. This was my attempt to tell the compiler don't worry about that.

Reply:
After accepting your change the build fails now get the following compiler error:
```
extension/SerialClient.ts(49,30): error TS7006: Parameter 'element' implicitly has an 'any' type.
```
This is why I originally put any there.

Reply:
Your latest commit with the updated typings fixed CI so it seems that was too due to outdated typings.